board.cpp

```diff
@@ -133,26 +234,55 @@ void Board::moveTile(const QModelIndex &tile)
             }
         }
 
-        emit dataChanged(m_selectedTileIndex, m_selectedTileIndex);
-        emit dataChanged(tile, tile);
-        //        popTiles(tile);
-        //        popTiles(m_selectedTileIndex);
+
+        if (popTiles(tile))
+        {
+            fillEmpty();
+        }
+        if (popTiles(m_selectedTileIndex))
+        {
+            fillEmpty();
+        }
```

Comment:
first we call both `popTiles` then `fillEmpty()`
New generated Tiles should not affect result of the previous move